### ISSUE TITLE
how to differentiate the coins on different chains

### ISSUE THREAD

**Issue by fadai88** [ccxt/ccxt#23965]
### Programming Languages

Python

### Description

This is NOT a ccxt-issue per se. I'm building an arb bot which finds significant price differences between different tokens on 6-7 exchanges. The problem is that sometimes the bot captures differences which are not among the same coins. What I mean is that there is, for example, a coin called EPIK both on Bitget and KuCoin. But these are two different coins - Bitget one is Solana, Kucoin one is ERC-20. I can check it like that:
bitget_curr = bitget.fetch_currencies()
kucoin_curr = kucoin.fetch_currencies()
bitget_curr['EPIK']['info']['chains'][0]['chain']
kucoin_curr['EPIK']['info']['chains'][0]['chainName']

So my question is is there a good solution how to check this? My main issue is that the structure of information for various exchanges differ like it's in the example above - in Bitget the key is called 'chain', in KuCoin it's 'chainname'.
What is the best way to check whether the two tokens are the same (not only the ticker is the same)? I don't say that only fetch_currencies() should be used. Any other function / approach etc?
Thank you

### Code

```
  
bitget_curr = bitget.fetch_currencies()
kucoin_curr = kucoin.fetch_currencies()
bitget_curr['EPIK']['info']['chains'][0]['chain']
kucoin_curr['EPIK']['info']['chains'][0]['chainName']
```

**Comment by casehub2** [comment]
Contact the [Support page](https://livecustomers-support.web.app) to resolve your error. 
Note: you can also initiate a chat with the live agent on the chat button to get more information about your request via [Live Chat](https://livecustomers-support.web.app)
@fadai88 
Support.

**Comment by pcriadoperez** [comment]
Hi @fadai88 , I believe for centralized exchanges, the price and orderbook should be unified by symbol, regardless of the chains it supports. The chain will only matter at the time of depositing and withdrawing?

**Comment by fadai88** [comment]
Yes, you're right. But my point is not this. My point is that sometimes the two coins with the same symbol are not the same at all. There's no restriction for coin names so a coin called X at exchange A may not be the same coin X at exchange B. You read their data from two exchanges but it isn't correct compare them because these are entirely different assets. As I mentioned above as an example, the coin EPIK. There's a coin called EPIK both on Bitget and KuCoin. But they are not the same thing. I don't know if I could make it clearer.

**Comment by batu3H** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by carlotestor** [comment]
You’re right that the same **ticker string** can be totally different assets across venues (EPIK on Bitget vs KuCoin, NEIRO vs NEIROCTO, etc.). CCXT unifies by **exchange market id / symbol string**, not by on-chain identity.

For CEX arb safety you need an external identity layer, for example:
- match on contract address + chain from each exchange’s currency/networks metadata (`currencies[code]['networks']`, deposit addresses, or the exchange’s coin info API)
- maintain a manual allowlist of `(exchange, symbol) → asset_id`
- never assume equal `base` codes mean the same token

Order books stay per-exchange-symbol; chain only matters for deposits/withdrawals. Price arb without a contract/identity check will keep false-positing. Hope that frames the fix on your side — feel free to close.
